src/webots/gui/WbGuiApplication.cpp

```diff
@@ -122,6 +122,75 @@ void WbGuiApplication::restart() {
 #endif
 }
 
+void WbGuiApplication::parseStreamArguments(const QString &serverArgument) {
+  bool monitorActivity = false;
+  bool disableTextStreams = false;
+  bool ssl = false;
+  bool controllerEdit = false;
+  int port = 1234;
+  QString mode = "x3d";
+
+#ifdef __APPLE__
+  const QStringList &options = argument.split(';', QString::SkipEmptyParts);
+#else  //  Qt >= 5.15
+  const QStringList &options = serverArgument.split(';', Qt::SkipEmptyParts);
+#endif
+  foreach (QString option, options) {
+    option = option.trimmed();
+    const QRegExp rx("(\\w+)\\s*=\\s*([A-Za-z0-9:/.\\-,]+)?");
+    rx.indexIn(option);
+    const QStringList &capture = rx.capturedTexts();
+    // "key" without value case
+    if (option == "monitorActivity")
+      monitorActivity = true;
+    else if (option == "disableTextStreams")
+      disableTextStreams = true;
+    else if (option == "ssl")
+      ssl = true;
+    else if (option == "controllerEdit")
+      controllerEdit = true;
+    else if (capture.size() == 3) {
+      const QString &key = capture[1];
+      const QString &value = capture[2];
+      if (key == "port") {
+        bool ok;
+        const int tmpPort = value.toInt(&ok);
+        if (ok)
+          port = tmpPort;
+        else {
+          cout << tr("webots: invalid option: '%1' in --stream").arg(serverArgument).toUtf8().constData() << endl;
+          cout << tr("webots: stream port has to be integer").toUtf8().constData() << endl;
+          mTask = FAILURE;
+        }
+      } else if (key == "mode") {
+        if (value != "x3d" && value != "mjpeg") {
+          cout << tr("webots: invalid option: '%1' in --stream").arg(serverArgument).toUtf8().constData() << endl;
+          cout << tr("webots: stream mode can only be x3d or mjpeg").toUtf8().constData() << endl;
+          mTask = FAILURE;
+        } else
+          mode = "mjpeg";
+      } else {
+        cout << tr("webots: unknown option: '%1' in --stream").arg(serverArgument).toUtf8().constData() << endl;
+        mTask = FAILURE;
+      }
+    } else {
+      cout << tr("webots: unknown option: '%1' in --stream").arg(serverArgument).toUtf8().constData() << endl;
+      mTask = FAILURE;
+    }
+  }
+  if (mTask != FAILURE) {
```

Comment:
Just very minor detail: this could be simplified by first checking if `mTask == FAILURE` and returning in the `if` block.